oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleDBRecord.java

```diff
@@ -50,31 +61,208 @@ protected SchemaReader getSchemaReader() {
+          stmt.setObject(sqlIndex, timestampWithTimeZone);
+        } else {
+          stmt.setNull(sqlIndex, sqlType);
+        }
+        break;
+      case OracleSchemaReader.BFILE:
+        if (field != null && record.get(field.getName()) != null) {
+          // Set value of Oracle 'BFILE' data type as instance of 'oracle.sql.BFILE'.
+          // Note, that we create only locator (link) to an external binary file (file stored outside of the database)
+          // and not content of the file.
+          Object value = record.get(field.getName());
+          byte[] bytes = value instanceof ByteBuffer ? Bytes.toBytes((ByteBuffer) value) : (byte[]) value;
+          Object bfile = createOracleBfile(stmt.getConnection(), bytes);
+          stmt.setObject(sqlIndex, bfile);
+        } else {
+          stmt.setNull(sqlIndex, sqlType);
+        }
+        break;
+      default:
+        super.writeToDB(stmt, field, fieldIndex);
+    }
+  }
+
+  /**
+   * Creates an instance of 'oracle.sql.TIMESTAMPTZ' which corresponds to the specified timestamp with time zone string.
+   * @param connection sql connection.
+   * @param timestampString timestamp with time zone string, such as "2019-07-15 15:57:46.65 GMT".
+   * @return instance of 'oracle.sql.TIMESTAMPTZ' which corresponds to the specified timestamp with time zone string.
+   */
+  private Object createOracleTimestampWithTimeZone(Connection connection, String timestampString) {
+    try {
+      ClassLoader classLoader = connection.getClass().getClassLoader();
+      Class<?> timestampTZClass = classLoader.loadClass("oracle.sql.TIMESTAMPTZ");
+      return timestampTZClass.getConstructor(Connection.class, String.class).newInstance(connection, timestampString);
+    } catch (ClassNotFoundException e) {
+      throw new InvalidStageException("Unable to load 'oracle.sql.TIMESTAMPTZ'.", e);
+    } catch (InstantiationException | InvocationTargetException | NoSuchMethodException | IllegalAccessException e) {
+      throw new InvalidStageException("Unable to instantiate 'oracle.sql.TIMESTAMPTZ'.", e);
+    }
+  }
 
+  /**
+   * Creates an instance of 'oracle.sql.BFILE'. Note, that we can not create an operating system file that a 'BFILE'
+   * would refer to. <a href="https://docs.oracle.com/cd/B19306_01/java.102/b14355/oralob.htm#BABJJEIC">Those
+   * are created only externally.</a>
+   * @param connection sql connection.
+   * @param bytes BFILE locator's bytes.
+   * @return instance of 'oracle.sql.BFILE'.
+   */
+  private Object createOracleBfile(Connection connection, byte[] bytes) {
+    try {
+      ClassLoader classLoader = connection.getClass().getClassLoader();
+      Class<?> oracleConnectionClass = classLoader.loadClass("oracle.jdbc.OracleConnection");
+      Class<?> bfileClass = classLoader.loadClass("oracle.sql.BFILE");
+      return bfileClass.getConstructor(oracleConnectionClass, byte[].class).newInstance(connection, bytes);
```

Comment:
It seems like a bad idea to create a BFile since the javadoc says it's deprecated and explicitly mentions to use OracleBfile instead, which has no way to be instantiated. I think we should just not support BFile for the sink.

Reply:
Removed. Sink changed to not support `BFILE` data type.